### PR TITLE
Cadence helm chart: Fix service annotations + Ingress YAML parsing

### LINKED ISSUE #990: Cadence helm chart's cadence-web service annotations are not documented and have YAML parsing errors
**Describe the bug**
`web-service.yaml` template of the `cadence` helm chart wrongly parses the service annotations here: https://github.com/banzaicloud/banzai-charts/blob/c651a7200ec5084889cbffa7df87380bc7a93e06/cadence/templates/web-service.yaml#L16

This is because `{{-` removes whitespace and results in wrong YAML indentation.

**Steps to reproduce the issue:**
Add service annotations such as the following in a custom `values.yaml` file and then use `helm template`:

```yaml
web:
  ingress:
    enabled: true
  service:
    type: 'NodePort'
    annotations:
      'beta.cloud.google.com/backend-config': '{"ports":{"http":"cloud-armor-backendconfig"}}'
```

**Expected behavior**
Helm template should output valid `service.yaml` file. Insted, a YAML parsing error is shown.

Furthermore, the `web-service`'s annotations are not documented in the chart's README file.

### PR DESCRIPTION
**Pull request by AgrimPrasad**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | yes
| API breaks?     | no|yes
| Deprecations?   | yes
| Related tickets | fixes #990 and #991 
| License         | Apache 2.0

### What's in this PR?
1. Document the cadence helm chart's `web-service` annotations.
1. Fix bug with YAML parsing of the cadence helm chart's `web-service` annotations.
1. Fix bug with YAML parsing of the cadence helm chart's `web-ingress` `apiVersion`.
1. Use updated `apiVersion` for the cadence helm chart's `web-ingress`

### Checklist

- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

**Comment by AgrimPrasad** [comment]
@sagikazarmark @ahma @kozmagabor @tarokkk Please see who would be the appropriate person to review this as we currently have to use the fork of the cadence helm chart due to these issues, thanks.

File: cadence/templates/web-ingress.yaml [hunk 1]
```diff
@@ -1,7 +1,5 @@
 {{- if .Values.web.ingress.enabled -}}
-{{- $serviceName := include "cadence.fullname" . -}}-web
-{{- $servicePort := .Values.web.service.port -}}
-apiVersion: extensions/v1beta1
+apiVersion: networking.k8s.io/v1beta1
```
Comment (sagikazarmark) [inline cadence/templates/web-ingress.yaml]: Can you please add a conditional check if the API version is available?

Comment (AgrimPrasad) [reply cadence/templates/web-ingress.yaml]: Thanks for the review @sagikazarmark 

I've added a conditional check for the `apiVersion` now. Please check again.

**Comment by AgrimPrasad** [comment]
I wonder what is the process of publishing the latest version of the chart to chartmuseum at https://kubernetes-charts.banzaicloud.com/ ?

**Comment by sagikazarmark** [comment]
I've been waiting for another bugfix to land in Cadence, but I guess it's blocking you, so here it is: https://github.com/banzaicloud/banzai-charts/releases/tag/cadence%2F0.5.0

Should be deployed to the chart repo soon.

**Comment by AgrimPrasad** [comment]
Thanks!

**Pull request merged**